pkg/sdn/plugin/pod_linux.go

```diff
@@ -182,28 +182,36 @@ func addMacvlan(netns string) error {
 		return fmt.Errorf("failed to find default route interface")
 	}
 
-	return ns.WithNetNSPath(netns, func(ns.NetNS) error {
-		err := netlink.LinkAdd(&netlink.Macvlan{
-			LinkAttrs: netlink.LinkAttrs{
-				MTU:         defIface.Attrs().MTU,
-				Name:        "macvlan0",
-				ParentIndex: defIface.Attrs().Index,
-			},
-			Mode: netlink.MACVLAN_MODE_PRIVATE,
-		})
-		if err != nil {
-			return fmt.Errorf("failed to create macvlan interface: %v", err)
-		}
-		l, err := netlink.LinkByName("macvlan0")
-		if err != nil {
-			return fmt.Errorf("failed to find macvlan interface: %v", err)
-		}
-		err = netlink.LinkSetUp(l)
-		if err != nil {
-			return fmt.Errorf("failed to set macvlan interface up: %v", err)
-		}
-		return nil
+	ns, err := ns.GetNS(netns)
+	if err != nil {
+		return fmt.Errorf("could not open netns %q", netns)
+	}
+	defer ns.Close()
+
+	err = netlink.LinkAdd(&netlink.Macvlan{
+		LinkAttrs: netlink.LinkAttrs{
+			MTU:         defIface.Attrs().MTU,
+			Name:        "macvlan0",
+			ParentIndex: defIface.Attrs().Index,
```

Comment:
You can give LinkAdd() the netns like so:

```
Namespace:   netlink.NsFd(int(ns.Fd())),
```

and it'll get put into the namespace after it's created.  Then I'd do "ns.Do()" for the LinkByName and LinkSetUp.